qt/applications/workbench/workbench/widgets/settings/model.py

```diff
@@ -6,19 +6,52 @@
 # SPDX - License - Identifier: GPL - 3.0 +
 #  This file is part of the mantid workbench
 import ast
+from typing import List, Dict, Any
 
 from mantid import ConfigService
 from workbench.config import CONF
-from workbench.widgets.settings.general.presenter import GeneralProperties
+from workbench.widgets.settings.general.general_settings_model import GeneralUserConfigProperties
+from workbench.widgets.settings.base_classes.config_settings_changes_model import ConfigSettingsChangesModel
 
 
 class SettingsModel:
+    def __init__(self, category_setting_models: List[ConfigSettingsChangesModel] = None):
+        self.category_setting_models: List[ConfigSettingsChangesModel] = category_setting_models
+        self.properties_which_need_a_restart = []
+
+    def register_property_which_needs_a_restart(self, property_string: str) -> None:
+        self.properties_which_need_a_restart.append(property_string)
+
+    def potential_changes_that_need_a_restart(self) -> List[str]:
+        def search_for_property_in_all_model_changes(property_string: str) -> bool:
+            for model in self.category_setting_models:
+                if property_string in model.properties_to_be_changed():
+                    return True
+            return False
```

Comment:
Can this function be declared outside the class, or is it declared here for a reason?

Reply:
just thought it was neater since it's only used here. I think it c++ I would've just made a lambda inside the function rather than creating a new function elsewhere, so I was thinking along those lines.